app/src/main/java/org/dosomething/letsdothis/ui/MainActivity.java

```diff
@@ -18,8 +32,128 @@ protected void onCreate(Bundle savedInstanceState)
+                    HubFragment fragment = (HubFragment) getSupportFragmentManager()
+                            .findFragmentByTag(HubFragment.TAG);
+                    if(fragment == null)
+                    {
+                        fragment = HubFragment.newInstance();
+                    }
+
+                    replaceCurrentFragment(fragment, HubFragment.TAG);
+                }
+            }
+        });
+
+        invites = findViewById(R.id.invites);
+        invites.setOnClickListener(new View.OnClickListener()
+        {
+            @Override
+            public void onClick(View v)
+            {
+                if(! invites.isSelected())
+                {
+                    InvitesFragment fragment = (InvitesFragment) getSupportFragmentManager()
+                            .findFragmentByTag(InvitesFragment.TAG);
+                    if(fragment == null)
+                    {
+                        fragment = InvitesFragment.newInstance();
+                    }
+
+                    replaceCurrentFragment(fragment, InvitesFragment.TAG);
+                }
+            }
+        });
+
+        notifications = findViewById(R.id.notifications);
+        notifications.setOnClickListener(new View.OnClickListener()
+        {
+            @Override
+            public void onClick(View v)
+            {
+                if(! notifications.isSelected())
+                {
+                    NotificationsFragment fragment = (NotificationsFragment) getSupportFragmentManager()
+                            .findFragmentByTag(NotificationsFragment.TAG);
+                    if(fragment == null)
+                    {
+                        fragment = NotificationsFragment.newInstance();
+                    }
+
+                    replaceCurrentFragment(fragment, NotificationsFragment.TAG);
+                }
+            }
+        });
+    }
+
+    private void replaceCurrentFragment(Fragment fragment, String tag)
+    {
+        if(! getSupportFragmentManager().popBackStackImmediate(tag, 0))
+        {
+            getSupportFragmentManager().beginTransaction().addToBackStack(tag)
+                                       .replace(R.id.container, fragment, tag).commit();
+            updateNavBar();
         }
     }
 
+    @Override
+    public void onBackStackChanged()
+    {
+        updateNavBar();
+    }
+
+    private void updateNavBar()
+    {
+        Fragment currentFragment = getSupportFragmentManager().findFragmentById(R.id.container);
+        String currentFragTag = currentFragment.getTag();
+
+        actions.setSelected(TextUtils.equals(ActionsFragment.TAG, currentFragTag));
+        notifications.setSelected(TextUtils.equals(NotificationsFragment.TAG, currentFragTag));
+        hub.setSelected(TextUtils.equals(HubFragment.TAG, currentFragTag));
+        invites.setSelected(TextUtils.equals(InvitesFragment.TAG, currentFragTag));
+
+    }
+
+    public static Intent getLaunchIntent(Context context)
```

Comment:
I would prefer to have this be called the same in all activites behave the similarly. I have called it callMe(), placed it at the top the class and also have the startActivity() code there. Lets discuss